lib/puppet/transaction/persistence.rb

```diff
@@ -87,7 +87,16 @@ def load
 
   # Save data from internal class to persistence store on disk.
   def save
-    Puppet::Util::Yaml.dump(@new_data, Puppet[:transactionstorefile])
+    converted_data = Puppet::Pops::Serialization::ToDataConverter.convert(
+      @new_data, {
+        rich_data: Puppet.lookup(:rich_data),
+        symbol_as_string: true,
```

Comment:
Why not keeps strings? won't this make the tests pass?

Reply:
Ah, won't works as the value is formated by https://github.com/puppetlabs/puppet/blob/main/lib/puppet/pops/serialization/to_data_converter.rb#L89-L96

And is returned as a String